federationapi/routing/send.go

```diff
@@ -558,7 +555,7 @@ func (t *txnReq) processEventWithMissingState(ctx context.Context, e *gomatrixse
 	// event ids and then use /event to fetch the individual events.
 	// However not all version of synapse support /state_ids so you may
 	// need to fallback to /state.
-
+	t.servers = t.getServers(ctx, e.RoomID())
```

Comment:
Is this really the first place where servers is unpopulated? I'm not convinced we don't need to call `getServers` in more places. I would just have the `t.servers` cache in `getServers` and if it's nil (which it shouldn't legit be) then do the lookup, rather than call `getServers` here and using `t.servers` elsewhere which I feel may result in uninitialised usage.

Reply:
Done.